Moving RadialProgress to the public repo

bholmes opened the pull request.

bholmes committed:
Moving RadialProgress to the public repo

mattleibow commented on XPlat/RadialProgress/component/Manifest.xml:
```diff
@@ -0,0 +1,12 @@
+<addon format="1">
```
should this file be here?

mattleibow commented on XPlat/RadialProgress/nuget/Xamarin.RadialProgress.nuspec:
```diff
@@ -0,0 +1,20 @@
+    <owners>Xamarin Inc.</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>A stylish, circular progress view for your Xamarin iOS and Android applications</description>
+    <copyright>Copyright 2016 Xamarin</copyright>
```
is this correct? I have been using "Copyright © Xamarin. All Rights Reserved." which I think I copied from MapBox

bholmes replied:
I have asked, and I have not been given good answers.  It is my understanding that it is not wrong.  Leaving as Xamarin.

mattleibow commented on XPlat/RadialProgress/nuget/Xamarin.RadialProgress.nuspec:
```diff
@@ -0,0 +1,20 @@
+    <title>Xamarin Radial Progress</title>
+    <version>1.0.2</version>
+    <authors>Xamarin Inc.</authors>
+    <owners>Xamarin Inc.</owners>
```
What should the `<authors>` and `<owners>` be for legal or whatnot.. I have just copied mapbox's "Xamarin"

bholmes replied:
Either MSFT or Xamarin is fine in this case.  Because this is not a change in features, just an oss release, I am deciding to keep Xamarin.

mattleibow commented on XPlat/RadialProgress/samples/RadialProgress.Android.Sample/MainActivity.cs:
```diff
@@ -0,0 +1,76 @@
+			btn = FindViewById<Button> (Resource.Id.myButton);
+			btn.Click += HandleClick;
+
+            bigRadialProgress.LabelTextDelegate = (val) => Math.Floor(100-val).ToString().PadLeft(2, '0');
```
whitespace ;)

mattleibow commented on XPlat/RadialProgress/samples/RadialProgress.Android.Sample/RadialProgress.Android.Sample.csproj:
```diff
@@ -0,0 +1,70 @@
+    <WarningLevel>4</WarningLevel>
+    <ConsolePause>False</ConsolePause>
+    <AndroidUseSharedRuntime>False</AndroidUseSharedRuntime>
+    <AndroidSupportedAbis>armeabi;armeabi-v7a;x86</AndroidSupportedAbis>
```
are these the abis that we want?

mattleibow commented on XPlat/RadialProgress/samples/RadialProgress.iOS.Sample/SampleViewController.cs:
```diff
@@ -0,0 +1,134 @@
+
+			// Add our different styles of RadialProgressViews
+            bigRadialProgressView = new RadialProgressView ();
+            bigRadialProgressView.LabelTextDelegate = (val) => Math.Floor(100-val*100).ToString().PadLeft(2, '0');
```
whitespace

mattleibow commented on XPlat/RadialProgress/samples/RadialProgress.iOS.Sample/SampleViewController.cs:
```diff
@@ -0,0 +1,134 @@
+			smallRadialProgressView.AutoresizingMask = UIViewAutoresizing.FlexibleMargins;
+			View.AddSubview (smallRadialProgressView);
+
+            tinyRadialProgressView = new RadialProgressView (progressType: RadialProgressViewStyle.Tiny);
```
whitespace

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.Android/RadialProgressView.cs:
```diff
@@ -0,0 +1,345 @@
+			get { return Value == MaxValue; }
+		}
+
+        public RadialProgressView (Context context, IAttributeSet attributes) : base(context, attributes)
```
whitespace

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.Android/RadialProgressView.cs:
```diff
@@ -0,0 +1,345 @@
+			LabelHidden = parser.LabelHidden;
+		}
+
+        public RadialProgressView (Context context, float minValue = 0f, float maxValue = 1f, RadialProgressViewStyle progressType = RadialProgressViewStyle.Big, Func<float, string> labelTextFunc = null) : base(context)
```
whitespace

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.iOS/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,33 @@
+//      Revision
+//
+[assembly: AssemblyVersion("1.0.1.0")]
+[assembly: AssemblyFileVersion("1.0.1.0")]
```
these version numbers should be "1.0.0.0"

bholmes replied:
maybe, but it is more important that they do not change.

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.iOS/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,33 @@
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("Xamarin Inc.")]
+[assembly: AssemblyProduct("RadialProgress")]
+[assembly: AssemblyCopyright("Copyright © Xamarin Inc. 2013")]
```
this may need updating

bholmes replied:
Not at this time

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.iOS/RadialProgressView.cs:
```diff
@@ -0,0 +1,253 @@
+        /// Gets or sets the label text delegate. If null, default function will be used (in %%)
+        /// </summary>
+        /// <value>The label text delegate.</value>
+        public Func<nfloat, string> LabelTextDelegate { get; set; }
```
whitespaces

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.iOS/RadialProgressView.cs:
```diff
@@ -0,0 +1,253 @@
+		public RadialProgressView (Func<nfloat, string> labelText = null, RadialProgressViewStyle progressType = RadialProgressViewStyle.Big)
+		{
+			this.progressType = progressType;
+            LabelTextDelegate = labelText;
```
whitespace

mattleibow commented on XPlat/RadialProgress/source/RadialProgress.iOS/RadialProgressView.cs:
```diff
@@ -0,0 +1,253 @@
+			radialProgressLayer.SetNeedsDisplay ();
+			
+			if (!LabelHidden) {
+                percentageLabel.Text = (LabelTextDelegate == null) ? Math.Floor (percentage).ToString ().PadLeft (2, '0') : LabelTextDelegate(currentValue);
```
whitespace

mattleibow reviewed (changes requested):
Looks pretty good, except for a few small tweaks needed:
 - copyright text and dates
 - in some files there is a mix between tabs and space (better to fix them now)

bholmes committed:
[RadialProgress] Changes requested in PR review

 - Removed Manifest.xml file that was not needed
 - Android ABIs in sample fixed
 - Several whitespace (tabs vs. spaces) issues fixed
